src/node_printer_posix.cc

```diff
@@ -349,8 +445,21 @@ MY_NODE_MODULE_CALLBACK(PrintDirect)
         RETURN_EXCEPTION_STR("unsupported format type");
     }
     type_str = itFormat->second;
+
     int num_options = 0;
-    cups_option_t *options = NULL;
+    cups_option_t *options = (cups_option_t *)0;
+
+    v8::Local<v8::Array> props = print_options->GetPropertyNames();
+
+    for(unsigned int i = 0; i < props->Length(); i++) {
```

Comment:
please prefix ++/-- operators. Suffix operators creates a temp value where here is useless.